packages/devtools_app/lib/src/inspector/inspector_screen.dart

```diff
@@ -83,8 +83,11 @@ class InspectorScreenBodyState extends State<InspectorScreenBody>
     super.initState();
     ga.screen(InspectorScreen.id);
 
-    autoDispose(
-        serviceManager.onConnectionAvailable.listen(_handleConnectionStart));
+    autoDispose(serviceManager.onConnectionAvailable.listen((service) {
+      setState(() {
```

Comment:
this refactor seems to be in the wrong direction. setState calls should be as close as to where state is set as possible. what is the motivation for doing this?

Reply:
_handleConnectionStart is called immediately below (line 92) which is in initState. We should not be calling `setState` from `initState` and that is what was causing an exception to be thrown. 

We were also calling `setState` twice in `_handleConnectionStart`, which contained only synchronous code, so it seemed unnecessary to call it twice instead of once for the entire method.

Reply:
calling _handleConnectionStart immediately in initState is probably the real bug. I would suggest we defer it to run after the frame or figure out how to refactor further so initState isn't triggering controller business logic like this.

Reply:
Did some refactoring PTAL.